### DISCUSSION TITLE
IS MGET supported in a cluster?

### DISCUSSION THREAD

**Discussion by mark-e-hoffman** [apache/kvrocks#1932] [Q&A]
Can I use the MGET redis command on a cluster? I've read that this command is not always supported against a clustered database?

**Comment by git-hulk** [comment]
It should depend on whether the keys are cross-slots or not.

**Comment by mark-e-hoffman** [comment]
I spun a cluster and I can't use MGET so I'm back to using a pipeline,
which works but is a little slower.

I control the number of clients hitting the database. It seems that as I
increase the number of concurrent clients, the response times increase
nearly linearly. I'm running on a 8 cpu linux box with 16 workers
configured. I was expecting a little better throughput when running a load
test where all 10 clients are hitting the database constantly. Any
suggestions on tuning configuration parameters to improve concurrency?

Thanks!

**Reply by git-hulk** [reply]
@mark-e-hoffman Did you enable the transaction mode while running the pipeline? If yes, can disable the transaction in client and have a try again.

**Comment by mark-e-hoffman** [comment]
I'm using a rust library and by default, it does not execute with a MULTI
EXEC, which is referred to as 'atomic'. I made the change and I'm seeing
much improved performance. I will continue  to test. Thanks for the
suggestion!

**Comment by mark-e-hoffman** [comment]
Hi, Things are going great with testing with KVROCKS. I've been able to
scale a 10 node cluster with nearly a billion keys and super good query
performance.
I was looking at the docs on doing a cluster migration in case I needed to
increase the size of my cluster. I see the CLUSTERX MIGRATE command but I
was wondering if you a
step-by-step guide on how to increase the nodes in a running cluster, from
adding the new node all the way thru the process.

Thanks

**Reply by git-hulk** [reply]
Hi @mark-e-hoffman that's cool.

I can add this document next week. I'm also working on the next version of [kvrocks-controller](https://github.com/apache/kvrocks-controller/pull/165) which will be ready to be reviewed in a few days, I believe it will help to simplify this process.
